Add distributed option deploy (#6186)
wazuh/wazuh-documentation: source/deployment-options/deploying-with-puppet/wazuh-puppet-module/index.rst

Install a stack via Puppet
--------------------------

Single Node:
------------

A single node stack of Wazuh manager, Wazuh dashboard, Wazuh indexer and Filebeat can be deployed using this manifest. See the Wazuh manager section for variables that can be used to configure the manager before deployment.

Create a stack.pp file at ``/etc/puppetlabs/code/environments/production/manifests/`` and put the contents below. ``puppet-agent-node`` refers to the hostname or IP of the puppet agent:

> .. code-block:: console

Reviewer: ```suggestion
.. code-block:: puppet
```

      $discovery_type = 'single-node'
      stage { 'certificates': }
      stage { 'repo': }
      stage { 'indexerdeploy': }
      stage { 'securityadmin': }
      stage { 'dashboard': }
      stage { 'manager': }
      Stage[certificates] -> Stage[repo] -> Stage[indexerdeploy] -> Stage[securityadmin] -> Stage[manager] -> Stage[dashboard]
      Exec {
      timeout => 0,
      }
      node "puppet-server" {
      class { 'wazuh::certificates':
        indexer_certs => [['node-1','127.0.0.1']],
        manager_certs => [['master','127.0.0.1']],
        dashboard_certs => ['127.0.0.1'],
        stage => certificates,
      }
      }
      node "puppet-aio-node" {
      class { 'wazuh::repo':
      stage => repo,
      }
      class { 'wazuh::indexer':
        stage => indexerdeploy,
      }
      class { 'wazuh::securityadmin':
      stage => securityadmin
      }
      class { 'wazuh::manager':
        stage => manager,
      }
      class { 'wazuh::filebeat_oss':
        stage => manager,
      }
      class { 'wazuh::dashboard':
        stage => dashboard,
      }
      }

Multi Node:
-----------

> You can deploy a distributed stack with 1 ``Wazuh dashboard``, 1 ``Wazuh manager master``, 1 ``Wazuh manager worker`` and 3 ``Wazuh indexer`` nodes between 3 diferentes servers or VMs with this manifest.

Reviewer: ```suggestion
With the multi-node manifest below, you can deploy a distributed stack with the following nodes between three different servers or Virtual Machines (VM).

-  3 indexer nodes
-  Manager master node
-  Manager worker node
-  Dashboard node
```

> It is required to add within the manifest the IPs of the servers where each of the applications will be installed.

Reviewer: ```suggestion
Within the manifest, include the IP addresses of the servers where you are installing each application.
```

This is the correspondence of the IPs with the puppet nodes described in the manifest:

> .. code-block:: console

Reviewer: What is this? These are not terminal commands, right? We can use `none` instead

Reviewer 2: Done: https://github.com/wazuh/wazuh-documentation/pull/6186/commits/3448299e91dcbc588dc4812a8b724b9deecd7bd5

    puppet-wazuh-indexer-node1 = node1host (Wazuh indexer node1)
    puppet-wazuh-indexer-node2 = node2host (Wazuh indexer node2)
    puppet-wazuh-indexer-node3 = node3host (Wazuh indexer node3)
    puppet-wazuh-manager-master = masterhost (Wazuh manager master)
    puppet-wazuh-manager-worker = workerhost (Wazuh manager worker)
    puppet-wazuh-dashboard = dashboardhost (Wazuh dashboard node)

> The wazuh::certificates class must be executed inside the puppet server where the Wazuh module is installed (``puppet-server``) because we use the archives module to pass files to all the servers where we deploy the Wazuh stack.

Reviewer: All these clarifications should go after the manifest code block below, once the reader have taken a look at it

Reviewer 2: Done fedb46d57

> The wazuh::certificates class must be executed inside the puppet server where the Wazuh module is installed (``puppet-server``) because we use the archives module to pass files to all the servers where we deploy the Wazuh stack.

Reviewer: ```suggestion
The ``wazuh::certificates`` class must be executed inside the puppet server where the Wazuh module is installed (``puppet-server``) because we use the archives module to pass files to all the servers where we deploy the Wazuh stack.
```

> If you need to add more ``Wazuh Indexer`` nodes, you need to add new variables (``indexer_node4_name`` and ``node4host`` for example) and add them to the ``indexer_discovery_hosts``, ``indexer_cluster_initial_master_nodes``, ``indexer_cluster_CN`` and ``indexer_certs`` arrays. In addition, it is necessary to add a new node instance equal to ``puppet-wazuh-indexer-node2`` or ``puppet-wazuh-indexer-node3``, so that they do not run ``securityadmin``.

Reviewer: ```suggestion
If you need more ``Wazuh Indexer`` nodes, add new variables, for example ``indexer_node4_name`` and ``node4host``. Add them to the following arrays:

-  ``indexer_discovery_hosts``
-  ``indexer_cluster_initial_master_nodes``
-  ``indexer_cluster_CN``
-  ``indexer_certs``

In addition, you need to add a new node instance similar to ``puppet-wazuh-indexer-node2`` or ``puppet-wazuh-indexer-node3``. These instances don't run ``securityadmin``.
```

> In case it is necessary to add any ``Wazuh manager worker`` server, it is necessary to add a new ``worker2host`` variable for example, add this variable to the ``manager_worker_certs`` array as ``['worker',"$worker2host"]`` and then replicate the node instance ``puppet-wazuh-manager-worker`` with the new server.

Reviewer: ```suggestion
In case you need to add a ``Wazuh manager worker`` server, add a new variable such as ``worker2host``. Add this variable to the ``manager_worker_certs`` array as ``['worker',"$worker2host"]`` and then replicate the node instance ``puppet-wazuh-manager-worker`` with the new server.
```
